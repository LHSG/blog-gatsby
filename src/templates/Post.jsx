import React from 'react';
import PropTypes from 'prop-types';

import { Head, Post } from '../components';
import {
  markdownRemark as markdownRemarkPropType,
  site as sitePropType,
} from '../proptypes';

export default function PostTemplate({
  data: {
    site,
    markdownRemark: {
      excerpt,
      frontmatter: { title, date, excerpt: frontmatterExcerpt, path, cover },
      html,
    },
  },
}) {
  const imageSizes = cover && cover.childImageSharp.sizes;
  const imageSrc = imageSizes && imageSizes.src;
  return (
    <div>
      <Head
        title={title}
        excerpt={frontmatterExcerpt || excerpt}
        path={path}
        site={site}
        image={imageSrc}
      />
      <Post
        title={title}
        date={date}
        html={html}
        excerpt={frontmatterExcerpt}
        coverImageSizes={imageSizes}
      />
    </div>
  );
}

PostTemplate.propTypes = {
  data: PropTypes.shape({
    markdownRemark: markdownRemarkPropType,
    site: sitePropType.isRequired,
  }).isRequired,
};

export const pageQuery = graphql`
  query PostByPath($path: String!) {
    ...SiteFragment
    markdownRemark(frontmatter: { path: { eq: $path } }) {
      ...MarkdownMetadataFragment
      ...MarkdownFrontmatterWithCoverFragment
    }
  }
`;
