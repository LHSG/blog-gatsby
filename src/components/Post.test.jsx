import React from 'react';
import { shallow } from 'enzyme';

import Post from './Post';
import { imagePropType } from '../../test/fixtures';

test('rendering', () => {
  const props = {
    title: 'title',
    date: 'date',
    excerpt: 'excerpt',
    html: '<div>some html</div>',
    coverImageSizes: imagePropType,
  };
  const wrapper = shallow(<Post {...props} />);
  expect(wrapper).toMatchSnapshot();
});
