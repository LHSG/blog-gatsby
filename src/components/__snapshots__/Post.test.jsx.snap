// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering 1`] = `
<div>
  <styled.h1>
    title
  </styled.h1>
  <styled.p
    dangerouslySetInnerHTML={
      Object {
        "__html": "excerpt",
      }
    }
  />
  <styled.p>
    <small>
      date
    </small>
  </styled.p>
  <Styled(Image)
    alt="title"
    sizes={
      Object {
        "sizes": "something",
      }
    }
  />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<div>some html</div>",
      }
    }
  />
  <hr />
  <p>
    <GatsbyLink
      to="/about"
    >
      Iâ€™m
    </GatsbyLink>
     Jia Hao, and I write software in San Francisco. Follow me on
     
    <a
      href="https://twitter.com/jiahaog"
      rel="noopener noreferrer"
    >
      Twitter
    </a>
    !
  </p>
</div>
`;
